app/controllers/measurements_controller.rb

```diff
@@ -2,47 +2,50 @@ class MeasurementsController < ApplicationController
   layout false
 
   def show
-    iteration_time = reference_time - params[:iteration].to_i * interval
-    cache_key = {
-      id: params[:id],
-      day: iteration_time.yday,
-      interval_number: interval_number(iteration_time)
-    }
-
-    @readings = Rails.cache.fetch(cache_key, expires_in: 25.hours) do
-      @measurement = Measurement.find(params[:id])
-      @measurement.
-        readings.
-        includes(:sensor).
-        where('time > ? AND time <= ?', from, to).
-        to_json(include: :sensor)
+    if params[:iteration].blank?
+      last_reading = Reading.where(measurement_id: params[:id]).last
+      readings = readings_json((last_reading&.time || Time.now),
+                               (2 * interval).minutes)
+    else
+      to = reference_time - (params[:iteration].to_i * interval).minutes
+      readings = readings_json(to, 25.hours)
     end
 
-    render json: @readings
+    render json: readings
   end
 
   private
 
+  def readings_json(to, expires_in)
+    cache_key = { id: params[:id], to: to&.to_f }
+
+    Rails.cache.fetch(cache_key, expires_in: expires_in) do
+      Reading.includes(:sensor).
+        where(measurement_id: params[:id]).
+        where('time > ? AND time <= ?', to - interval.minutes, to).
+        to_json(only: :value, include: { sensor: { only: [:lat, :long] } })
+    end
+  end
+
   def interval_number(time)
-    minutes = (time.seconds_since_midnight/60).to_i
+    minutes = (time.getutc.seconds_since_midnight/60).to_i
```

Comment:
This is nice overview about timezones in rails http://www.elabs.se/blog/36-working-with-time-zones-in-ruby-on-rails